src/HotReload/FSProcess.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace SwooleTW\Http\HotReload;
+
+use Swoole\Process as SwooleProcess;
+use Symfony\Component\Process\Process as AppProcess;
+
+/**
+ * Class FSProcess
+ */
+class FSProcess
+{
+    /**
+     * Observed files.
+     *
+     * @var array
+     */
+    protected $filter;
+
+    /**
+     * Watch recursively.
+     *
+     * @var bool
+     */
+    protected $recursively;
+
+    /**
+     * Watch directory.
+     *
+     * @var string
+     */
+    protected $directory;
+
+    /**
+     * When locked event cannot do anything.
+     *
+     * @var bool
+     */
+    protected $locked;
+
+    /**
+     * FSProcess constructor.
+     *
+     * @param string $filter
+     * @param bool $recursively
+     * @param string $directory
+     */
+    public function __construct(string $filter, bool $recursively, string $directory)
+    {
+        $this->filter = $filter;
+        $this->recursively = $recursively;
+        $this->directory = $directory;
+        $this->locked = false;
+    }
+
+    /**
+     * Make swoole process.
+     *
+     * @param callable|null $callback
+     *
+     * @return \Swoole\Process
+     */
+    public function make(?callable $callback = null)
+    {
+        $mcb = function ($type, $buffer) use ($callback) {
+            if (! $this->locked && AppProcess::OUT === $type && $event = FSEventParser::toEvent($buffer)) {
```

Comment:
I'm not sure why the event I got here is always `null`. So the hot reload didn't get triggered after receiving a fswatch event. Do you have any ideas?

Reply:
I will see, what happens. What is you use?

Reply:
I'm using OSX and `fswatch 1.14.0`.

Reply:
> I'm using OSX and `fswatch 1.14.0`.

I'm too...

I fix it.